QuizApp/App/Repository/QuizRepository.swift

```diff
@@ -0,0 +1,62 @@
+protocol QuizRepositoryProtocol {
+
+    func fetchQuizzes() async -> [QuizRepositoryModel]
+
+    func fetchQuizzes(for category: CategoryModel) async -> [QuizRepositoryModel]
+
+    func getQuestions(for quizId: Int) async -> [QuestionRepositoryModel]
+
+    func saveQuizzes(_ quizzes: [QuizDatabaseModel])
+
+}
+
+final class QuizRepository: QuizRepositoryProtocol {
+
+    private let networkDataSource: QuizNetworkDataSourceProtocol
+    private let databaseDataSouce: QuizDatabaseDataSourceProtocol
+
+    init(networkDataSource: QuizNetworkDataSourceProtocol, databaseDataSouce: QuizDatabaseDataSourceProtocol) {
+        self.networkDataSource = networkDataSource
+        self.databaseDataSouce = databaseDataSouce
+    }
+
+    @MainActor
+    func fetchQuizzes() async -> [QuizRepositoryModel] {
+        do {
+            let quizzes =  try await networkDataSource.fetchQuizes()
```

Comment:
Can you check on which thread are `networkDataSource` functions executed.

And you have one extra space.

Reply:
This is the reason I used `@MainActor`.

`*** Terminating app due to uncaught exception 'RLMException', reason: 'Realm accessed from incorrect thread.'`

Instances of Realm and Object are thread-contained. They cannot be passed between threads or that exception will occur. So with `@MainActor,` instances are on the same thread. These are the threads used without `@MainActor`.

```
Thread on: <NSThread: 0x6000020a48c0>{number = 7, name = (null)}
Thread on: <NSThread: 0x6000020e11c0>{number = 4, name = (null)}
```